src/test/java/AmazonTest.java

```diff
@@ -85,16 +85,16 @@ public void dealsOfTodayPageFunctionality() {
     }
 
     @Test
-    public void checkPriceFilterFunctionality()  {
+    public void checkPriceFilterFunctionality() {
         openHomePageAndChooseDealsOfToday();
         DealsOfTodayPage dealsOfTodayPage = new DealsOfTodayPage(CommonActions.getDriver());
-        dealsOfTodayPage.clickTheElementThatHasTheFollowingText("Women's Fashion from Daily Ritual");
+        dealsOfTodayPage.clickOnTheNameOfTheSelectedSpecies();
         FashionOfWomenFromRitualPage fashionOfWomenFromRitualPage = new FashionOfWomenFromRitualPage(CommonActions.getDriver());
         fashionOfWomenFromRitualPage.waitForPageLoad();
-        fashionOfWomenFromRitualPage.filterPrice("Up to $25");
+        fashionOfWomenFromRitualPage.clickOnFilterPrice();
```

Comment:
So on which filter does your method click? Does it contain the exact functionality that it does?